packet.go

```diff
@@ -66,6 +67,98 @@ type PacketAdaptationExtensionField struct {
 	SpliceType             uint8  // Indicates the parameters of the H.262 splice.
 }
 
+//ParsePacket parses a packet into
+func ParsePacket(b []byte) (p *Packet, err error) {
+	return parsePacket(astikit.NewBytesIterator(b))
+}
+
+//ParsePSIPacket parses a known PSI packet
+func ParsePSIPacket(p *Packet) (*PSIData, error) {
+	return parsePSIData(astikit.NewBytesIterator(p.Payload))
+}
+
+//ParsePESPacket parses a known PES packet
+func ParsePESPacket(p *Packet) (d *PESData, err error) {
+	defer func() {
+		if re := recover(); re != nil {
+			err = fmt.Errorf("Error parsing packet: %v", re)
+		}
+	}()
```

Comment:
Is there risk for a panic in all of these functions, or specifically these two? Would it be worth adding this to all of these new functions?

Reply:
I don't think so, can probably be removed.

Reply:
Ah no that's right. These ones are necessary as we might get a header that is referring to a packet that is yet to arrive. Eg the packet is 400 bytes long, but we only have the first 188 of it. So this will stop us from failing directly - it isn't a problem but we need to protect against it here.

Reply:
Ah okay. Is there no way of knowing in advance how much of the packet we have?

Reply:
It uses a custom iterator type from github.com/asticode/go-astikit which I don't think has length checks atm.

Reply:
On inspection it does, so I removed these and my test cases still pass - so I have removed them.